docs/concepts/node-agent.md

```diff
@@ -0,0 +1,59 @@
+
+## Architecture
+
+![Design](./images/gardener-nodeagent-architecture.drawio.svg)
+
+This figure visualizes the overall architecture of the `gardener-node-agent`. It starts with the operating system config (OSC) being transferred through the userdata to a machine through the `machine-controller-manager` (MCM). The bootstrap phase of the `gardener-node-agent` will then happen as described in the previous section.
+
+## Reasoning
+
+The `gardener-node-agent` is a replacement for what was called the `cloud-config-downloader` and the `cloud-config-executor`, both written in `bash`. The `gardener-node-agent` implements this functionality as a regular controller and feels more uniform in terms of maintenance.
+
+With the new Architecture we gain a lot, let's describe the most important gains here.
+
+### Developer Productivity
+
+Because we all develop in go day by day, writing business logic in `bash` is difficult, hard to maintain, almost impossible to test. Getting rid of almost all `bash` scripts which are currently in use for this very important part of the cluster creation process will enhance the speed of adding new features and removing bugs.
+
+### Speed
+
+Until now, the `cloud-config-downloader` runs in a loop every 60sec to check if something changed on the shoot which requires modifications on the worker node. This produces a lot of unneeded traffic on the api-server and wastes time, it will sometimes take up to 60sec until a desired modification is started on the worker node.
+By using the controller-runtime we can watch for the `node`, the `OSC` in the `secret`, and the shoot-access-token in the `secret`. If any of these object changed, and only then, the required action will take effect immediately.
+This will speed up operations and will reduce the load on the api-server of the shoot especially for large clusters.
+
+## Scalability
+
+The `cloud-config-downloader` add a random wait time before restarting the `kubelet` in case the `kubelet` was updated or a configuration change was made to it. This is required to reduce the load on the API server and the traffic on the internet uplink. It also reduces the overall downtime of the services in the cluster because every `kubelet` restart takes a node for several seconds into `NotReady` state which eventually interrupts service availability.
+
+```
+TODO: The `gardener-node-agent` could do this in a much more intelligent way because it watches the `node` object. The gardenlet could add some annotation which tells the `gardener-node-agent` to wait for the kubelet in a coordinated manner. The coordination could be in chunks of nodes and wait for them to finish and then start with the next chunk. Also a equal time spread is possible.
+```
```

Comment:
Why not add this in a section future work/outlook or something similar? It helps understand what we want to achieve now and what might come later.

Reply:
Added a `Future Development` section and improved the wording.